### DISCUSSION TITLE
Problem with rem units when migrating to v6

### DISCUSSION THREAD

**Discussion by johansunden** [mantinedev/mantine#3788] [Q&A] [answered]
**Context**
- At my company we have a 2 year old project with a lot of code
- 6 months ago we started introducing Mantine as our main component library
- From the beginning we've had font-size: 0.625em specified in the html element in our global stylesheet
- Just like Mantine decided for v6, we wanted to make our UI scalable and accessibly by using rem units rather than px
- We found that a common trick to make it easier to work with rem units was to specify 0.625em on the html element to make 1 rem = 10px, rather than 1 em = 16px

When we started using Mantine we realized we were now mixing rem and px units, since a lot of Mantine was done with px, but it is what it is we figured, it works well enough. But now that Mantine v6 is switching to using rem units we have a problem of course. I want/need to keep the html element's 0.625em in order to not break all of our non-Mantine UI, but at the same time I can't have Mantine components be based on 10px instead of 16px because then everything becomes super small.

**Question**
So hopefully you understand my problem. So my question would be, what is the best approach at solving this?

ChatGTP told me I could set some --mantine-font-size-base, but I don't think such a variable exist, and I'm not sure if that suggestion even made sense. 

Reading the documentation I would think that maybe the best approach would be to manually set the **fontSizes**, **radius** and **spacing** on the Theme object in my MantineProvider? Maybe also the font-size on **headings** **h1**-**h6** too? Maybe also **shadows** and **breakpoints**? Basically convert the rem values there from ones that assume a baseline of 16px to ones that assume a baseline of 10px, producing the same result given the browser font-size setting?

I haven't tried much yet, I would like to hear what you guys think is my best remedy here.

 And also if setting all of these values would actually solve the issue for me, or are there like "hard-coded" rem values in various components? Which would mean that bits and pieces of various Mantine components would be off, even if I change all of these rem values?

I there no way, or have you had any thoughts about, making the baseline configurable? Meaning that I could tell Mantine that my baseline is 10px rather than the default 16px. This would mean, as an example, that some place in the Mantine code that have specified 1rem would change to say 1.6rem with baseline set to 10px.

**Comment by rtivital** [comment, marked as answer]
Mantine does not support changing baseline – 1rem is always 16px.

> And also if setting all of these values would actually solve the issue for me, or are there like "hard-coded" rem values in various components?

Yes, there are lots of such values.

> I there no way, or have you had any thoughts about, making the baseline configurable?

It is also not planned to change this in the future as it will introduce another dynamic part to the styles which should be avoided.
In your case, I would advise continuing using 5.x version for old projects and use 6.x version for new projects that do not have 10px baseline.

**Comment by azhard4int** [comment]
We faced a similar issue while upgrading our app Usermaven to 6.0.2. Unfortunately, we had to roll back to version 5. It would be great if we can use the baseline configuration.

**Comment by johansunden** [comment]
We ended up using a script that converted all our rem values like 'Xrem' to be a sass function: rem(X*10). So now instead of our original baseline of 10px we have a baseline of 16px, but can use this sass function in our styles, just like Mantine have an equivalent function in JS, to convert pixel values to the appropriate rem value! So we managed to upgrade after all.
